### PR TITLE
Automatically retrieve the remote OS major ver (SOFTWARE-3729)

### PR DESCRIPTION
**Pull request by brianhlin**
Instead of using the env var

### CODE REVIEW

File: 30-remote-site-setup.sh [hunk 1]
```diff
@@ -41,20 +43,18 @@ foreach_bosco_endpoint setup_ssh_config
...
 setup_endpoints_ini () {
     remote_home_dir=$(ssh -i $BOSCO_KEY ${ruser}@${rhost} pwd)
+    remote_os_ver=$(ssh -i $BOSCO_KEY ${ruser}@${rhost} "rpm -E %rhel")
     cat <<EOF >> $ENDPOINT_CONFIG
 [Endpoint ${RESOURCE_NAME}-${ruser}]
 local_user = ${ruser}
 remote_host = ${rhost}
 remote_user = ${ruser}
 remote_dir = $remote_home_dir/bosco-osg-wn-client
+upstream_url = https://repo.opensciencegrid.org/tarball-install/3.4/osg-wn-client-latest.${remote_os_ver}.x86_64.tar.gz
```
Comment (edquist) [inline 30-remote-site-setup.sh]:
```suggestion
upstream_url = https://repo.opensciencegrid.org/tarball-install/3.4/osg-wn-client-latest.el${remote_os_ver}.x86_64.tar.gz
```

So, `rpm -E %rhel` just returns the int, eg `6` or `7`, but these tarballs are named (eg) `osg-wn-client-latest.el6.x86_64.tar.gz`

**Comment by edquist** [comment]
Regarding `set -xe` -

I take it you do actually want all the commands printed to stderr, so in that case i think it should work as expected.  But i guess beware that this makes a lot of noise for a script.

Also i notice you now turn explicitly them off at the end of the scripts (with `set +xe`).  Is that because the these scripts are `source`d?  Because otherwise if you run a standalone bash script that `set`s these options, they do not affect the calling script (that is, their effect won't go beyond the end of the script where they appear).

**Comment by brianhlin** [comment]
Yup, these scripts are sourced so I only want the `set` statements to affect the given scripts (i.e. allow users to have their own scripts that run after these that are `set +xe`)

From my testing, `-x` doesn't actually dump the commands to the `supervisord_startup.err` file :(

**Comment by edquist** [comment]
> Yup, these scripts are sourced so I only want the `set` statements to affect the given scripts (i.e. allow users to have their own scripts that run after these that are `set +xe`)

Ok, makes sense.  Btw yes turns out i was wrong about `bash -xe` not working; you were correct that you can include any `set` options in the `bash` command line.  However, since you are sourcing them, i believe they'll only turn on if you specify the `set` command; the `bash -xe` in the `#!` line i think only works if you invoke it as a script.

> From my testing, `-x` doesn't actually dump the commands to the `supervisord_startup.err` file :(

that's a little bit surprising...

**Pull request merged**